src/main/java/app/servlets/AddRecordServlet.java

```diff
@@ -0,0 +1,190 @@
+package app.servlets;
+
+import app.dbService.DBException;
+import app.dbService.DBService;
+import app.dbService.dataSets.Book;
+import org.apache.tomcat.util.http.fileupload.FileItem;
+import org.apache.tomcat.util.http.fileupload.FileItemFactory;
+import org.apache.tomcat.util.http.fileupload.FileUploadException;
+import org.apache.tomcat.util.http.fileupload.disk.DiskFileItemFactory;
+import org.apache.tomcat.util.http.fileupload.servlet.ServletFileUpload;
+import org.apache.tomcat.util.http.fileupload.servlet.ServletRequestContext;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import org.apache.commons.codec.binary.Base64;
+import java.util.Iterator;
+import java.util.List;
+
+public class AddRecordServlet extends HttpServlet {
+    private DBService service = new DBService();
+    private Book book;
+    private String UPLOAD_DIRECTORY;
+    private File file;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init();
+        book = new Book();
+        UPLOAD_DIRECTORY ="D:\\Book Library. ItechArt\\src\\main\\webapp\\images";
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String action = req.getParameter("action");
+        req.setAttribute("book", book);
+        switch (action == null ? "info" : action) {
+            case "update":
+                req.getRequestDispatcher("views/full_book_page.jsp").forward(req, resp);
+                break;
+            case "info":
+            default:
+                req.getRequestDispatcher("views/book_page.jsp").forward(req, resp);
+                break;
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```

Comment:
this method is so long, I forgot what was in the beginning. Try to split it up into manageable parts